src/api/discovers.ts

```diff
@@ -1,17 +1,34 @@
 import { insertSupabase, TABLES } from 'services/supabase';
+import { hasLength } from 'utils/misc-utils';
+
+interface DiscoverRequest {
+    capture_name: string;
+    connector_tag_id: string;
+    draft_id: string;
+    endpoint_config: any;
+    update_only: boolean;
+    data_plane_name?: string;
+}
 
 export const discover = (
     entityName: string,
     config: any,
     connectorID: string,
     draftID: string,
-    updateOnly?: boolean
+    updateOnly?: boolean,
+    dataPlaneName?: string
 ) => {
-    return insertSupabase(TABLES.DISCOVERS, {
+    const data: DiscoverRequest = {
         capture_name: entityName,
         endpoint_config: config,
         connector_tag_id: connectorID,
         draft_id: draftID,
         update_only: updateOnly ?? false,
-    });
+    };
+
+    if (hasLength(dataPlaneName)) {
+        data.data_plane_name = dataPlaneName;
+    }
```

Comment:
I think you can just set the value in the object directly and then if it is undefined Supabase should remove the value - not 100% sure though.

Reply:
That was not the case when I attempted this early on.